_posts/2021-04-13-how-to-update-to-terraform-14.adoc

```diff
@@ -0,0 +1,227 @@
+excerpt: Learn how to update your infrastructure code that depends on the Gruntwork Infrastructure as Code Library to Terraform 0.14.
+tags: ["aws", "terraform", "terragrunt"]
+cloud: ["aws"]
+redirect_from: /static/guides/upgrades/how-to-update-to-terraform-14/
+---
+:page-type: guide
+:page-layout: post
+
+:toc:
+:toc-placement!:
+
+// GitHub specific settings. See https://gist.github.com/dcode/0cfbf2699a1fe9b46ff04c41721dda74 for details.
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+toc::[]
+endif::[]
+
+== Intro
+
+This guide will walk you through how to update any code that depends on the
+https://gruntwork.io/infrastructure-as-code-library/[Gruntwork Infrastructure as Code Library] to
+https://www.terraform.io/upgrade-guides/0-14.html[Terraform 0.14]. Terraform 0.14 introduces several new features
+and fixes, but it also has a number of backwards incompatibilities that have to be incorporated into your codebase.
+
+=== What you'll learn in this guide
+
+This guide consists of two main sections:
+
+<<core_concepts>>::
+  An overview of Terraform 0.14 and why it is important to update your code for compatibility.
+
+<<deployment_walkthrough>>::
+  The steps you need to take to update your code relying on the Gruntwork Infrastructure as Code library to work with
+  Terraform 0.14. Includes a
+  <<compatibility_table,version compatibility table>> you can use as a reference to know which Gruntwork Repo version
+  tag is compatible with Terraform 0.14.
+
+
+[[core_concepts]]
+== Core Concepts
+
+=== Background
+
+https://www.hashicorp.com/blog/announcing-hashicorp-terraform-0-14-general-availability[Terraform 0.14 was released on
+December 02, 2020]. Some of the major new features in 0.14 include:
+
+. Consise diff output in `plan`.
+. The ability to mark input variables as `sensitive`.
+. A lock file for provider dependency versions.
+
+For more info, check out the
+https://www.hashicorp.com/blog/announcing-hashicorp-terraform-0-14-general-availability[announcement blog post].
+
+0.14 is a major new release for Terraform, which means it includes some backwards incompatible changes. We have
+gone through all the Terraform modules in the https://gruntwork.io/infrastructure-as-code-library/[Gruntwork
+Infrastructure as Code Library] and updated them to work with Terraform 0.14. In order to use 0.14, you will need to
+update to these new versions and make other changes to your code, as described in the following section.
+
+
+
+[[deployment_walkthrough]]
+== Deployment walkthrough
+
+=== Step 1: update your code to be compatible with Terraform 0.13
+
+If you haven't already, you need to:
+
+. Update your code to work with Terraform 0.13. Do NOT skip from, say, 0.11, straight to 0.14. You MUST update to
+  0.13.0 or above first! If you're still on Terraform 0.11 or older, see our
```

Comment:
This notation "0.13.0" is confusing for me. Perhaps "0.13.x" is better?

Reply:
Why is 0.13.0 confusing? That's the first release on the 0.13.x line and is the minimum version allowable here...